Sources/SwiftInspectorVisitors/TypeDescription.swift

```diff
@@ -0,0 +1,238 @@
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+import SwiftSyntax
+
+public enum TypeDescription: Codable, Equatable, CustomStringConvertible {
+  case simple(name: String)
+  indirect case member(name: String, baseType: TypeDescription)
+  indirect case composition([TypeDescription])
+  indirect case optional(TypeDescription)
+  indirect case implicitlyUnwrapped(TypeDescription)
+  indirect case array(TypeDescription)
+  indirect case dictionary(key: TypeDescription, value: TypeDescription)
+  indirect case tuple([TypeDescription])
+  case unknown(text: String)
+
+  /// A description of this type that can be used for code-generation to represent this type.
+  public var description: String {
+    switch self {
+    case let .simple(name):
+      return name
+    case let .array(type):
+      return "[\(type)]"
+    case let .composition(types):
+      return types.map { $0.description }.joined(separator: " & ")
+    case let .optional(type):
+      return "\(type)?"
+    case let .implicitlyUnwrapped(type):
+      return "\(type)!"
+    case let .dictionary(key, value):
+      return "[\(key): \(value)]"
+    case let .member(name, baseType):
+      return "\(baseType).\(name)"
+    case let .tuple(types):
+      return "(\(types.map { $0.description }.joined(separator: ", ")))"
+    case let .unknown(text):
+      return text
+    }
+  }
+
+  public init(from decoder: Decoder) throws {
+    let values = try decoder.container(keyedBy: CodingKeys.self)
+    let caseDescription = try values.decode(String.self, forKey: .caseDescription)
+    if caseDescription == Self.simpleDescription {
+      let text = try values.decode(String.self, forKey: .text)
+      self = .simple(name: text)
+
+    } else if caseDescription == Self.unknownDescription {
+      let text = try values.decode(String.self, forKey: .text)
+      self = .unknown(text: text)
+
+    } else if caseDescription == Self.memberDescription {
+      let text = try values.decode(String.self, forKey: .text)
```

Comment:
I'm having trouble following why we use `.text` in some places and `.textDescription` in others.

Reply:
@bachand I don't think I ever use `textDescription`. Did you mean something else?

Reply:
ohhh `typeDescription`! Yeah I'll document the coding keys and that should make it more clear.

Reply:
https://github.com/fdiaz/SwiftInspector/pull/69#discussion_r569709703

Reply:
Ah yes, `typeDescription`. I'm glad you figured it out.